File: platforms/python/examples/core/renderer/Renderer.py
from rendercanvas.auto import RenderCanvas, loop

from fragmentcolor import Shader, Renderer

renderer = Renderer()

# Use your platform's windowing system to create a window
canvas = RenderCanvas(size=(800, 600))

# Create a Target from it
target = renderer.create_target(canvas)
texture_target = renderer.create_texture_target([16, 16])

# RENDERING
renderer.render(Shader(""), texture_target)

# That's it. Welcome to FragmentColor!

File: platforms/swift/examples/core/renderer/Renderer.swift

import FragmentColor

let renderer = Renderer()

// Use your platform's windowing system to create a window
// iOS: window/canvas provided by CAMetalLayer at runtime

// Create a Target from it
let target = try await renderer.createTextureTarget([800, 600])
let texture_target = try await renderer.createTextureTarget([16, 16])

// RENDERING
try renderer.render(Shader(""), texture_target)

// That's it. Welcome to FragmentColor!